### PR TITLE
DY-1 transformation engine infrastructure

### PR DESCRIPTION
**Pull request by spicy-sauce**

### CODE REVIEW

File: src/datayoga/__init__.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    Returns:
+        Job: Compiled job
+    """
+    logger.debug("Compiling job")
+    job = Job(job_yaml)
+    return job
+
+
+def transform(job: Job, data: Any, context: Context = None) -> Any:
```
Comment (zalmane) [inline src/datayoga/__init__.py]: I think we can remove this.
if you use dy.compile you get a Job and can then call its 'transform' method.
If we want 'transform' as dy.transform, it should receive the Yaml as a convenience to skip the 'compile' part separately.

File: src/datayoga/block.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        self.properties = properties
+        self.validate()
+        self.init()
+
+    def validate(self):
+        """
+        Validates block against its JSON Schema
+        """
+        json_schema_file = path.join(os.path.dirname(os.path.realpath(
+            sys.modules[self.__module__].__file__)), "block.schema.json")
```
Comment (zalmane) [inline src/datayoga/block.py:39]: does this work after we packaged it as a wheel?

File: src/datayoga/block.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        Args:
+            data (Any): Data
+
+        Returns:
+            Any: Transformed data
+        """
+        pass
+
+    def get_block_name(self):
+        return self.__class__.__name__
```
Comment (zalmane) [inline src/datayoga/block.py:79]: how does this get set? what is it used for?

File: src/datayoga/utils.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import json
+import re
+from typing import Any, Dict
+
+import yaml
+
+
+def snake_to_camel(value: str) -> str:
```
Comment (zalmane) [inline src/datayoga/utils.py]: where do we use camelCase?

File: src/datayoga/__init__.py (hunk 1 above)
Comment (spicy-sauce) [reply src/datayoga/__init__.py]: done

File: src/datayoga/block.py (hunk 3 above)
Comment (spicy-sauce) [reply src/datayoga/block.py:79]: used in the block implementation, for example:
```logger.debug(f"Initializing {self.get_block_name()}")```

File: src/datayoga/utils.py (hunk 4 above)
Comment (spicy-sauce) [reply src/datayoga/utils.py]: In the yaml file we use `remove_field` while the class itself named `RemoveField`

File: src/datayoga/block.py (hunk 2 above)
Comment (spicy-sauce) [reply src/datayoga/block.py:39]: yes

File: src/datayoga/__init__.py [hunk 5]
```diff
@@ -22,17 +22,18 @@ def compile(job_yaml: Dict[str, Any]) -> Job:
...
     Args:
-        job (Job): Job to transform against
+        job_yaml (Dict[str, Any]): Job in YAML format
         data (Any): Data to transform
         context (Context, optional): Context. Defaults to None.
 
     Returns:
         Any: Transformed data
     """
+    job = compile(job_yaml)
```
Comment (zalmane) [inline src/datayoga/__init__.py]: this should not be "yaml" if we are not reading the source itself. it is just a dict, so Json and YAML are the same here

Comment (spicy-sauce) [reply src/datayoga/__init__.py]: right, fixed

**Pull request merged**